### PR TITLE
Fixes pyprobml-review/pyprobml-review#22

### PR DESCRIPTION
**Pull request by nitish1295**
## Description

Added sensor_fusion_2d.ipynb for [sensor_fusion_2d.ipynb](https://github.com/probml/pyprobml/blob/master/notebooks/book1/03/sensor_fusion_2d.ipynb)

### Figure Number
Book 1 - 3.10

### Figures

Kindly note, that for now, I have added the figures here since it is easier to keep track and review. Will create a gist for master repo

| Before PR |image |
| :-: | :-: |
| After PR | image|

### Issue
- Pyprobml Issue https://github.com/probml/pyprobml/issues/739

## Steps

- [x] Changed the name of the variables/functions to PEP 8 style
- [x] Added code to generate three figures as given in the book and Latexified them.
- [x] Converted code to JAX
- [x] Added interactive demo to change sigma for Spherical and Unequal case

## Checklist:

- [x] Performed a self-review of the code
- [x] Tested on Google Colab.

## Potential problems/Important remarks

- Looking for suggestion for axes labels
- ~~Already checking with Zeel regarding the inconsistencies in type hinting since JAX dev suggests using ```Any```, refer https://github.com/google/jax/issues/10186. I am not sure about the information conveyed if we use```Any```. For all other parameters I have included type hints~~. Checked with Zeel, will add type hints for JAX as jnp.ndarray
-  As discussed with Zeel no titles for the graphs have been added

### CODE REVIEW

File: notebooks/book1/03/sensor_fusion_2d.ipynb [hunk 1]
```diff
@@ -0,0 +1,503 @@
+{
```
Comment (nitish1295) [inline notebooks/book1/03/sensor_fusion_2d.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #46.        # Code below is unused.
</pre><p>Code below was unused, let me know if this can be removed?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (patel-zeel) [reply notebooks/book1/03/sensor_fusion_2d.ipynb:1]: <p>Sure</p>

Comment (patel-zeel) [inline notebooks/book1/03/sensor_fusion_2d.ipynb:1]: <p>Isn't the font size too small? was this creating a problem? Ideally we should start with default 9 and iteratively decrease it if it is creating problems.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (patel-zeel) [inline notebooks/book1/03/sensor_fusion_2d.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #68.                    "label": "$E[x|y_1; y_2]$",
</pre><p>This should use E[x|y_1, y_2] as per the book.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/31/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (nitish1295) [reply notebooks/book1/03/sensor_fusion_2d.ipynb:1]: Yes will fix that

Comment (nitish1295) [reply notebooks/book1/03/sensor_fusion_2d.ipynb:1]: Yea I agree it is small but given the way the graph was organized, I just managed to tuck it in the upper left corner using the font_size=6. I will try to see if I can maybe make it slightly larger

**Comment by patel-zeel** [comment]
@nitish1295 Once you address the remaining comments, you can go ahead and open a PR on the main repo without another round of review.

**Comment by karm-patel** [comment]
Thanks @nitish1295, I've modified your code and added latexified figure in book :)

**Comment by nitish1295** [comment]
@karm-patel Thanks for this!!
